extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+			$fakeUser->setRealName( '' );
+			$fakeUser->setName( $this->mOldUsername );
+			$fakeUser->addToDatabase();
+
+			$fakeUser->setGlobalAttribute( 'renameData', self::RENAME_TAG . '=' . $this->mNewUsername . ';' . self::PROCESS_TAG . '=' . '1' );
+			$fakeUser->setGlobalFlag( 'disabled', 1 );
+
+			try {
+				$fakeUser->setPassword( null );
+			} catch ( PasswordError $e ) {
+				// We don't really care if the password wasn't set at all for a fake user
+				// now that we go through Helios
+			}
+
+			$fakeUser->saveSettings();
+			$this->mFakeUserId = $fakeUser->getId();
+			$this->addLog( "Created fake user account for {$fakeUser->getName()} with ID {$this->mFakeUserId} and renameData '{$fakeUser->getGlobalAttribute( 'renameData', '')}'" );
+		} else {
+			$this->addLog( "Fake user account already exists: {$this->mFakeUserId}" );
+		}
+
+		$this->invalidateUser( $this->mOldUsername );
+
+		// process global tables
+		$this->addLog( "Initializing update of global shared DB's." );
+		$this->updateGlobal();
+
+		$callParams = array(
+			'requestor_id' => $this->mRequestorId,
+			'requestor_name' => $this->mRequestorName,
+			'rename_user_id' => $this->mUserId,
+			'rename_old_name' => $this->mOldUsername,
+			'rename_new_name' => $this->mNewUsername,
+			'rename_fake_user_id' => $this->mFakeUserId,
+			'phalanx_block_id' => $this->mPhalanxBlockId,
+			'reason' => $this->mReason,
+			'notify_renamed' => $this->mNotifyUser,
+		);
+		$task = ( new UserRenameTask() )
+			->setPriority( \Wikia\Tasks\Queues\PriorityQueue::NAME );
+		$task->call( 'renameUser', $wikiIDs, $callParams );
+		$this->mUserRenameTaskId = $task->queue();
+
+		return true;
+	}
+
+	/**
+	 * Processes shared database (wikicities) and makes all needed changes
+	 */
+	public function updateGlobal() {
+		// wikicities
+		$this->addLog( "Updating global shared database: wikicities." );
+		$dbw = WikiFactory::db( DB_MASTER );
+		$dbw->begin();
+		$tasks = self::$mGlobalDefaults;
+
+		$hookName = 'UserRename::Global';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, array( $dbw, $this->mUserId, $this->mOldUsername, $this->mNewUsername, $this, &$tasks ) );
+
+		foreach ( $tasks as $task ) {
+			$this->addLog( "Updating {$task['table']}.{$task['username_column']}." );
+			$this->renameInTable( $dbw, $task['table'], $this->mUserId, $this->mOldUsername, $this->mNewUsername, $task );
+		}
+
+		$hookName = 'UserRename::AfterGlobal';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, array( $dbw, $this->mUserId, $this->mOldUsername, $this->mNewUsername, $this, &$tasks ) );
+
+		$dbw->commit();
+		$this->addLog( "Finished updating shared database: wikicities." );
+	}
+
+	/**
+	 * Processes specific local wiki database and makes all needed changes
+	 *
+	 * Important: should only be run within maintenance script (bound to specified wiki)
+	 *
+	 * @throws DBError
+	 */
+	public function updateLocal() {
+		global $wgCityId, $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app